### PR TITLE
Add OCP Service Runtime configuration template

### PR DESCRIPTION
**Pull request by redhatrises**
#### Description:

- Adds OVAL template for checking `ps -ef` ocp process configuration options. For example,

```
$ ps -ef | grep hyperkube | grep anonymous-auth

root     15243     1  5 Sep24 ?        1-00:38:17 /usr/bin/hyperkube kubelet --v=2 --address=0.0.0.0 --allow-privileged=true --anonymous-auth=true --authentication-token-webhook=true --authentication-token-webhook-cache-ttl=5m --authorization-mode=Webhook --authorization-webhook-cache-authorized-ttl=5m --authorization-webhook-cache-unauthorized-ttl=5m --bootstrap-kubeconfig=/etc/origin/node/bootstrap.kubeconfig --cadvisor-port=0 --cert-dir=/etc/origin/node/certificates --cgroup-driver=systemd --client-ca-file=/etc/origin/node/client-ca.crt --cluster-dns=10.10.93.6 --cluster-domain=cluster.local --container-runtime-endpoint=/var/run/dockershim.sock --containerized=false --enable-controller-attach-detach=true --experimental-dockershim-root-directory=/var/lib/dockershim --fail-swap-on=false --feature-gates=RotateKubeletClientCertificate=true,RotateKubeletServerCertificate=true --file-check-frequency=0s --healthz-bind-address= --healthz-port=0 --host-ipc-sources=api --host-ipc-sources=file --host-network-sources=api --host-network-sources=file --host-pid-sources=api --host-pid-sources=file --hostname-override= --http-check-frequency=0s --image-service-endpoint=/var/run/dockershim.sock --iptables-masquerade-bit=0 --kubeconfig=/etc/origin/node/node.kubeconfig --max-pods=250 --network-plugin=cni --node-ip= --node-labels=node-role.kubernetes.io/master=true --pod-infra-container-image=registry.access.redhat.com/openshift3/ose-pod:v3.10.34 --pod-manifest-path=/etc/origin/node/pods --port=10250 --read-only-port=0 --register-node=true --root-dir=/var/lib/origin/openshift.local.volumes --rotate-certificates=true --tls-cert-file= --tls-cipher-suites=TLS_ECDHE_ECDSA_WITH_CHACHA20_POLY1305 --tls-cipher-suites=TLS_ECDHE_RSA_WITH_CHACHA20_POLY1305 --tls-cipher-suites=TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256 --tls-cipher-suites=TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256 --tls-cipher-suites=TLS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384 --tls-cipher-suites=TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384 --tls-cipher-suites=TLS_ECDHE_ECDSA_WITH_AES_128_CBC_SHA256 --tls-cipher-suites=TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA256 --tls-cipher-suites=TLS_ECDHE_ECDSA_WITH_AES_128_CBC_SHA --tls-cipher-suites=TLS_ECDHE_ECDSA_WITH_AES_256_CBC_SHA --tls-cipher-suites=TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA --tls-cipher-suites=TLS_ECDHE_RSA_WITH_AES_256_CBC_SHA --tls-cipher-suites=TLS_RSA_WITH_AES_128_GCM_SHA256 --tls-cipher-suites=TLS_RSA_WITH_AES_256_GCM_SHA384 --tls-cipher-suites=TLS_RSA_WITH_AES_128_CBC_SHA --tls-cipher-suites=TLS_RSA_WITH_AES_256_CBC_SHA --tls-min-version=VersionTLS12 --tls-private-key-file=
```

### CODE REVIEW

File: shared/templates/create_ocp_service_runtime_config.py [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+import re
+
+from template_common import FilesGenerator, UnknownTargetError
+
+
+class OCPServiceRuntimeConfigGenerator(FilesGenerator):
+    def generate(self, target, ocp_process_info):
+        process_cmd, process_cmd_option, process_cmd_val = ocp_process_info
+        # convert variable name to a format suitable for 'id' tags
+        ocp_proc_id = re.sub('[-\._]', '_', process_cmd_option.strip("--="))
```
Comment (jan-cerny) [inline shared/templates/create_ocp_service_runtime_config.py]: There shouldn't be a `\` and the regexes are usually prefixed by r in Python.

File: shared/templates/create_ocp_service_runtime_config.py [hunk 2]
```diff
@@ -0,0 +1,53 @@
+#
+# create_ocp_service_runtime_config.py
+#   automatically generate checks for ocp procuess runtime configurations
```
Comment (jan-cerny) [inline shared/templates/create_ocp_service_runtime_config.py]: procuess

File: shared/templates/template_OVAL_ocp_service_runtime_config [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+  <unix:process58_test check="all" comment="ocp process runtime parameter {{{ OCPCMDOPTION }}} set to {{{ OCPCMDVAL }}}" id="test_ocp_runtime_{{{ OCPCMDOPTIONID }}}" version="1">
+    <unix:object object_ref="object_ocp_runtime_{{{ OCPCMDOPTIONID }}}" />
+    <unix:state state_ref="state_ocp_runtime_{{{ OCPCMDOPTIONID }}}" />
+  </unix:process58_test>
+  <unix:process58_object id="object_ocp_runtime_{{{ OCPCMDOPTIONID }}}" version="1">
+    <unix:command_line operation="pattern match">^{{{ OCPPROCESS }}}.*$</unix:command_line>
+    <unix:pid datatype="int" operation="greater than">0</unix:pid>
+  </unix:process58_object>
+  <unix:process58_state id="state_ocp_runtime_{{{ OCPCMDOPTIONID }}}" version="1">
+    <unix:command_line operation="pattern match">^.*[\s]+{{{ OCPCMDOPTION }}}={{{ OCPCMDVAL }}}([\s]+.*$|$)</unix:command_line>
```
[the remark this answers is not in the document]
Comment (shawndwells) [reply shared/templates/template_OVAL_ocp_service_runtime_config]: Yes, and a good question! While the configuration files sometimes use spaces, this template evaluates the process arguments which have key=value syntax.

File: ocp3/templates/csv/ocp_service_runtime_config.csv [hunk 4]
```diff
@@ -0,0 +1 @@
+/usr/bin/hyperkube,--anonymous-auth,false
```
[the remark this answers is not in the document]
Comment (jan-cerny) [reply ocp3/templates/csv/ocp_service_runtime_config.csv]: I'm afraid we don't have that now. I think it now requires that variables are defined in `.var` files. We would have to improve the build system to support generating variables from templates. That probably should be solved in general, so let's not do it in this PR.

**Comment by jan-cerny** [comment]
I've checked that it generates the OVAL definition. Thank you for your contribution.

**Pull request merged**